### PR TITLE
Inference validation flow

### PR DESCRIPTION
**Pull request by pixelplex**
Each finished inference that passes the probabilistic ShouldValidate check triggers validation on the instance that owns the escrow session. A Postgres lease row is inserted on acquisition (ON CONFLICT DO NOTHING as a safety net against accidental double-processing) and tracks the validation lifecycle via a status field: pending → submitted or skipped.

The validator fetches the prompt and response payloads from the executor, verifies the executor's signature and payload hashes, then re-runs the inference on the ML node. The result is added to the host's in-memory mempool as MsgValidation and delivered to the client in the next diff response. MsgValidation is a devshard session message, not a Cosmos transaction — there is no tx_hash to record.

The lease table is the recovery primitive. If an instance fails between acquiring the lease and submitting MsgValidation, the row stays pending. A RetryLoop scans active in-memory sessions every DEVSHARD_VALIDATION_RETRY_INTERVAL (default 5m) for leases that remained pending past DEVSHARD_VALIDATION_LEASE_TTL (default 30m), reassigns them with FOR UPDATE SKIP LOCKED, and reruns the full validation flow. Leases from a prior epoch are marked skipped — the client session that would consume the result is already gone. Lease rows are deleted on each epoch transition.

Integration tests cover the valid path (all ML nodes agree, settlement stats reflect completed validations), the invalid path (one node returns bad logits, inference is challenged), and the retry path (backdated lease is resubmitted by the loop within 60s at 1s interval).

### CODE REVIEW

**Comment by DimaOrekhovPS** [comment]
Integration tests: https://github.com/gonka-ai/gonka/actions/runs/25417459642

File: common/queryapi/epoch.go [hunk 1]
```diff
@@ -94,49 +93,79 @@ func (h *Handlers) getEpochParticipants(ctx context.Context, epoch uint64) (*gen
 		return nil, echo.NewHTTPError(http.StatusBadRequest, "Epoch enumeration starts with 1")
 	}
 
-	// Query ActiveParticipants from the chain store with merkle proof.
-	result, err := h.chain.CometServiceClient().ABCIQuery(ctx, &cmtservice.ABCIQueryRequest{
-		Path:  "store/inference/key",
-		Data:  inferencetypes.ActiveParticipantsFullKey(epoch),
-		Prove: true,
+	// First query without proof to read CreatedAtBlockHeight from the value.
```
Comment (DimaOrekhovPS) [inline common/queryapi/epoch.go:96]: Could you briefly explain the reason behind the changes? Just asking, because it's not obvious how it's related to inference validation

File: devshard/cmd/devshardd/config.go [hunk 2]
```diff
@@ -225,6 +242,19 @@ func envOr(key, fallback string) string {
...
+	v := os.Getenv(key)
+	if v == "" {
+		return fallback, nil
+	}
+	d, err := time.ParseDuration(v)
+	if err != nil {
+		return 0, fmt.Errorf("invalid duration %q: %w", v, err)
+	}
+	return d, nil
+}
```
[the remark this answers is not in the document]
Comment (DimaOrekhovPS) [reply devshard/cmd/devshardd/config.go:259]: Seems valid, please validate these parameters when constructing the retry loop

File: common/queryapi/epoch.go (hunk 1 above)
Comment (kstdl) [reply common/queryapi/epoch.go:96]: This change is here because AI review highlighted an existing queryapi proof correctness issue. I decided not to split it into a separate PR just for this small fix

File: devshard/cmd/devshardd/app.go [hunk 3]
```diff
@@ -217,6 +216,38 @@ func buildHostManager(
 	if err := manager.RecoverSessions(); err != nil {
 		slog.Warn("recover sessions failed", "error", err)
 	}
+
+	retryLoop := session.NewRetryLoop(leaseStore, validator, manager, phase, instanceAddr)
+	retryLoop.WithInterval(cfg.ValidationRetryInterval)
+	retryLoop.WithLeaseTTL(cfg.ValidationLeaseTTL)
+	go retryLoop.Run(ctx)
```
[the remark this answers is not in the document]
Comment (kstdl) [reply devshard/cmd/devshardd/app.go]: fixed

File: devshard/cmd/devshardd/app.go [hunk 4]
```diff
@@ -217,6 +216,38 @@ func buildHostManager(
...
+			}()
+		}
+
+		if currentEpoch >= 4 {
+			expiredPayloadEpoch := currentEpoch - 3
+			go func() {
+				if err := payloadStore.DropEpoch(context.Background(), expiredPayloadEpoch); err != nil {
+					slog.Warn("payload epoch cleanup failed", "error", err)
+				}
+			}()
```
[the remark this answers is not in the document]
Comment (kstdl) [reply devshard/cmd/devshardd/app.go]: fixed

File: devshard/cmd/devshardd/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kstdl) [reply devshard/cmd/devshardd/config.go:259]: fixed

**Review by DimaOrekhovPS — APPROVED** [review]
